android/engine/src/main/java/org/smartregister/fhircore/engine/ui/components/ActionableButton.kt

```diff
@@ -16,61 +16,136 @@
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
 import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
 import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
 import androidx.compose.material.ButtonDefaults
 import androidx.compose.material.Icon
 import androidx.compose.material.OutlinedButton
 import androidx.compose.material.Text
 import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.ArrowDropDown
 import androidx.compose.material.icons.filled.Check
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
-import org.smartregister.fhircore.engine.domain.model.ActionableButtonData
+import org.smartregister.fhircore.engine.configuration.view.ButtonProperties
+import org.smartregister.fhircore.engine.domain.model.ServiceStatus
 import org.smartregister.fhircore.engine.ui.theme.DangerColor
 import org.smartregister.fhircore.engine.ui.theme.DefaultColor
+import org.smartregister.fhircore.engine.ui.theme.InfoColor
+import org.smartregister.fhircore.engine.ui.theme.PersonalDataBackgroundColor
 import org.smartregister.fhircore.engine.ui.theme.SuccessColor
 
+const val ACTIONABLE_BUTTON_TEXT_TEST_TAG = "actionable_button_text_test_tag"
+const val ACTIONABLE_BUTTON_START_ICON_TEST_TAG = "actionableButtonStartIconTestTag"
+const val ACTIONABLE_BUTTON_END_ICON_TEST_TAG = "actionableButtonEndIconTestTag"
+const val ACTIONABLE_BUTTON_OUTLINED_BUTTON_TEST_TAG = "actionableButtonOutlinedButtonTestTag"
+
 @Composable
 fun ActionableButton(
-  actionableButtonData: ActionableButtonData,
+  buttonProperties: ButtonProperties,
   modifier: Modifier = Modifier,
-  onAction: (String, String?) -> Unit
+  onAction: () -> Unit
 ) {
   OutlinedButton(
-    onClick = {
-      if (actionableButtonData.questionnaireId != null)
-        onAction(
-          actionableButtonData.questionnaireId,
-          actionableButtonData.backReference?.reference
-        )
-    },
+    onClick = { if (buttonProperties.questionnaire?.id != null) onAction() },
     colors =
       ButtonDefaults.buttonColors(
-        backgroundColor = actionableButtonData.contentColor.copy(alpha = 0.1f),
-        contentColor = actionableButtonData.contentColor.copy(alpha = 0.9f)
+        backgroundColor = buttonProperties.statusColor().copy(alpha = 0.1f),
+        contentColor = buttonProperties.statusColor().copy(alpha = 0.1f)
```

Comment:
Any reason for changing the opacity for this from 0.9f to 0.1f?

Reply:
Thanks for this catch, I updated it for some testing with existing views, this is fixed now.